GeosCore/cldj_interface_mod.F90

```diff
@@ -1213,4 +1216,141 @@ FUNCTION Get_RH_Index( RH ) RESULT( index )
+!
+! !IROUTINE: CldJ_Dryrun_Check
+!
+! !DESCRIPTION: Checks if the CLOUD-J configuration files are present
+!  on disk or if they need to be downloaded by dry-run.
+!\\
+!\\
+! !INTERFACE:
+!
+  SUBROUTINE CloudJ_DryRun_Check( Input_Opt )
+!
+! USES:
+!
+    USE ErrCode_Mod
+    USE Input_Opt_Mod, ONLY : OptInput
+!
+! !INPUT PARAMETERS:
+!
+    TYPE(OptInput), INTENT(IN)  :: Input_Opt   ! Input Options object
+!
+! !REMARKS:
+!
+! !REVISION HISTORY:
+!  19 Dec 2023 - R. Yantosca - Initial version
+!  See https://github.com/geoschem/geos-chem for complete history
+!EOP
+!------------------------------------------------------------------------------
+!BOC
+    ! Scalars
+    LOGICAL            :: amIRoot
+
+    ! Strings
+    CHARACTER(LEN=255) :: fileName
+
+    !========================================================================
+    ! CloudJ_DryRun_Check begins here!
+    !========================================================================
+    amIRoot = Input_Opt%amIRoot
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'CJ77_inp.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'FJX_spec.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'FJX_scat-aer.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'FJX_scat-cld.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'FJX_scat-ssa.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'FJX_scat-UMa.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'FJX_scat-geo.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'atmos_std.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'atmos_h2och4.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'atmos_geomip.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+    fileName = TRIM( Input_Opt%CloudJ_Dir ) // 'FJX_j2j.dat'
+    CALL Check_File_For_DryRun( fileName, amIRoot )
+
+  END SUBROUTINE CloudJ_DryRun_Check
+!EOC
+!------------------------------------------------------------------------------
+!                  GEOS-Chem Global Chemical Transport Model                  !
+!------------------------------------------------------------------------------
+!BOP
+!
+! !IROUTINE: Check_File_For_DryRun
+!
+! !DESCRIPTION: Checks the status of a single file in dry-run simulations.
+!\\
+!\\
+! !INTERFACE:
+!
+ SUBROUTINE Check_File_for_DryRun( fileName, amIRoot )
+!
+! USES:
+!
+    USE ErrCode_Mod
+    USE Input_Opt_Mod, ONLY : OptInput
+!
+! !INPUT PARAMETERS:
+!
+    CHARACTER(LEN=*), INTENT(IN) :: fileName   ! Name of file to check
```

Comment:
Specify the name includes the full path. Otherwise it looks like only the name is passed which may lead to confusion.

Reply:
Now updated in 22c8094